pyspectral/rayleigh.py

```diff
@@ -151,75 +160,100 @@ def get_reflectance_lut(self):
         secant, azimuth difference angle, and sun zenith secant
 
         """
-
-        return get_reflectance_lut(self.reflectance_lut_filename)
+        if self._rayl is None:
+            lut_vars = get_reflectance_lut(self.reflectance_lut_filename)
+            self._rayl = lut_vars[0]
+            self._wvl_coord = lut_vars[1]
+            self._azid_coord = lut_vars[2]
+            self._satz_sec_coord = lut_vars[3]
+            self._sunz_sec_coord = lut_vars[4]
+        return self._rayl, self._wvl_coord, self._azid_coord,\
+            self._satz_sec_coord, self._sunz_sec_coord
 
     def get_reflectance(self, sun_zenith, sat_zenith, azidiff, bandname,
                         redband=None):
         """Get the reflectance from the three sun-sat angles."""
         # Get wavelength in nm for band:
         wvl = self.get_effective_wavelength(bandname) * 1000.0
-        rayl, wvl_coord, azid_coord, satz_sec_coord, sunz_sec_coord = self.get_reflectance_lut()
+        rayl, wvl_coord, azid_coord, satz_sec_coord, sunz_sec_coord = \
+            self.get_reflectance_lut()
+
+        # force dask arrays
+        compute = False
+        if HAVE_DASK and not isinstance(sun_zenith, Array):
+            compute = True
+            sun_zenith = from_array(sun_zenith, chunks=sun_zenith.shape)
+            sat_zenith = from_array(sat_zenith, chunks=sat_zenith.shape)
+            azidiff = from_array(azidiff, chunks=azidiff.shape)
+            if redband is not None:
+                redband = from_array(redband, chunks=redband.shape)
 
         clip_angle = np.rad2deg(np.arccos(1. / sunz_sec_coord.max()))
-        sun_zenith = np.clip(np.asarray(sun_zenith), 0, clip_angle)
+        sun_zenith = clip(sun_zenith, 0, clip_angle)
         sunzsec = 1. / np.cos(np.deg2rad(sun_zenith))
         clip_angle = np.rad2deg(np.arccos(1. / satz_sec_coord.max()))
```

Comment:
Yes. If you would like (and I wouldn't disagree) I can import all these methods from dask if installed. It's possible these only work because they come down to multiplying the number by something and the dask arithmetic methods catch that.

Reply:
Yes, I think it's preferable to avoid any bad surprises upon upgrading numpy in the future.

Reply:
Looks good!